### PR TITLE
Task/frontend 111 implement preferences modal

### PR DESCRIPTION
**Pull request by WanderningMaster**

### CODE REVIEW

**Comment by WanderningMaster** [comment]
[Implement preferences modal (for recommendations)](https://trello.com/c/7cwI41e2/111-implement-preferences-modal-for-recommendations)

File: frontend/src/components/common/preferences-modal/preferences-modal-container.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const { choosedPreferences } = useAppSelector((state) => ({
+    choosedPreferences: state.preference,
+  }));
+
+  const [isNeedModal, setIsNeedModal] = useState(false);
+  const [isNeedConfirmModal, setIsNeedConfirmModal] = useState(false);
+
+  useEffect(() => {
+    if (choosedPreferences.dataStatus === DataStatus.FULFILLED) {
+      setIsNeedModal(!choosedPreferences.data.length);
```
Comment (droztoff) [inline frontend/src/components/common/preferences-modal/preferences-modal-container.tsx]: boolean expected, cast this argument explicitly to boolean (const isEmptyPreferences = Boolean(choosedPreferences.data.length))

File: frontend/src/components/common/preferences-modal/preferences-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    dispatch(getCategories());
+  }, [dispatch]);
+
+  const handleClickCard = (id: string): void => {
+    setPickedCategories({
+      categories: pushOrDeleteIfExists<string>(id, pickedCategories.categories),
+    });
+  };
+
+  const isPicked = (id: string): boolean => !!pickedCategories.categories.filter((category) => category === id).length;
```
Comment (droztoff) [inline frontend/src/components/common/preferences-modal/preferences-modal.tsx]: use "!!" is bad practice

File: frontend/src/components/common/preferences-modal/preferences-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { Modal } from '../modal/modal';
+import { FC, useEffect } from 'react';
+import style from './styles.module.scss';
```
Comment (droztoff) [inline frontend/src/components/common/preferences-modal/preferences-modal.tsx]: name must be plural

**Review by droztoff — APPROVED** [review]

**Pull request merged**